lib/gtl_formatter.rb

```diff
@@ -125,10 +125,13 @@ def self.format_cols(view, row, controller, options)
                 :text  => text}.compact
       elsif COLUMN_WITH_TIME.include?(col)
         celltext = format_time_for_display(row, col)
-      elsif COLUMN_WITH_OS_ICON.key?(col)
-        @osicon = send(COLUMN_WITH_OS_ICON[col], record)
-        @ostext = format_col_for_display(view, row, col).presence || _("Unknown")
       elsif COLUMN_WITH_OS_TEXT.include?(col)
+        view.col_order.each do |column|
+          if COLUMN_WITH_OS_ICON.key?(column)
+            @osicon = send(COLUMN_WITH_OS_ICON[column], record)
+            @ostext = format_col_for_display(view, row, column).presence || _("Unknown")
+          end
+        end
```

Comment:
```suggestion
        osicon_col = view.col_order.detect { |c| COLUMN_WITH_OS_ICON.key?(column) }
        @osicon = send(COLUMN_WITH_OS_ICON[osicon_col], record)
        @ostext = format_col_for_display(view, row, osicon_col).presence || _("Unknown")

```

Reply:
And like I said above, if osicon and ostext are not used anywhere else outside of this code, then we can make them local variables and not instance variables.

I think they were made instance variables because they were discovered on different parts of the loop but needed to be shared.

Reply:
@Fryguy Fixed it like you suggested and replaced instance variables with local variables.Thank you.